instance/tests/test_tasks.py

```diff
@@ -146,13 +172,29 @@ def test_make_appserver_active(self, active, mock_make_active):
         tasks.make_appserver_active(self.appserver.id, active=active)
         mock_make_active.assert_called_once_with(active=active)
 
+    @ddt.data([False, True], [False, False])
+    @patch('instance.models.server.OpenStackServer.update_status')
+    @patch('instance.models.openedx_appserver.OpenEdXAppServer.make_active')
+    def test_make_appserver_active_server_unhealthy(self, health, mock_make_active, mock_update_status):
+        """
+        By default, we activate the appserver.
+        """
+        with patch('instance.models.server.OpenStackServer.Status.Ready.is_healthy_state',
+                   new_callable=PropertyMock) as mock_is_healthy_state:
+            mock_is_healthy_state.side_effect = health
+            tasks.make_appserver_active(self.appserver.id, active=True)
+            mock_update_status.assert_called_once_with()
+            if not all(not data for data in health):
+                mock_make_active.assert_called_once_with(active=True)
+
     @patch('instance.models.openedx_appserver.OpenEdXAppServer.make_active')
     def test_deactivate_others(self, mock_make_active):
         """
         When activating the appserver, optionally deactivate others.
         """
         for dummy in range(5):
             appserver = make_test_appserver(self.appserver.instance)
+            appserver.server = ReadyOpenStackServerFactory()
```

Comment:
I noticed we are adding this in **lot** of different places. This is bad: it's code repetition everywhere. Please move this to a common place, like `make_test_appserver`, so it's not done like this. You can add it as a keyword arg that'll default to the ready OpenStack server, or whichever way you think is best.